tests/resources/test_object_setters.py

```diff
@@ -53,7 +53,7 @@ def test_object_pointer_serde():
 def test_object_validation():
     """Test that an object pointing to another object is validated."""
     meas = MeasurementRun("A measurement")
-    with pytest.raises(TypeError):
+    with pytest.raises(ValueError):
```

Comment:
Same thing here as with LOE above.

Reply:
The problem here is that gemd-python (correctly) throws TypeErrors when passing arguments of the wrong type, and _serialization throws ValueErrors.  This change is reflective of the new MRO.  

It's even a little grosser than that, since the error message thrown is actually misleading.
```
ValueError: <class 'citrine.resources.measurement_run.MeasurementRun'> <Measurement run 'A measurement'> is not one of valid types: (<class 'dict'>, <class 'NoneType'>) for MaterialRun:process
```

Reply:
Ugh, I've seen that error message and it's awful.

Sounds like the underlying issue is that incongruence, and it will likely be more complicated to preserve as is now. Is `deserialize` an operation users will often call, or is it more an internally used method?

We should move `_serialization` to `TypeError` when we go to 3.0.